homeassistant/components/flux_led/__init__.py

```diff
@@ -96,12 +97,23 @@ async def _async_migrate_unique_ids(hass: HomeAssistant, entry: ConfigEntry) ->
 
     @callback
     def _async_migrator(entity_entry: er.RegistryEntry) -> dict[str, Any] | None:
-        # Old format {entry_id}.....
-        # New format {unique_id}....
+
+        assert unique_id is not None
```

Comment:
It is not a good idea to have an assert in production code, especially not without a log message.

Reply:
The assert is to let mypy know the variable isn't undefined since it can't figure it out

There is a check above that ensures we never get here if it is None
https://github.com/home-assistant/core/pull/68029/files#diff-d3985dc844a3779a721e969985806c63cd6d2c9ba28b9dfdd674f1fc32c2ae41R94

I could wrap it in `if TYPE_CHECKING:`, but I generally prefer not to do that since it
is an implementation error if it ever does fail the assertion and it would be good to blow up
so the mistake is obvious.

Reply:
`assert` won't blow up when running optimized.

When `assert` is needed, there is another issue in the design/error handling in general. We use it way too much in general IMHO.

Reply:
I refactored to remove the assert.  Its slightly less efficient, but `mypy` can figure it out now